fbw-a32nx/src/systems/shared/src/performance/a32nx_takeoff.ts

```diff
@@ -0,0 +1,3704 @@
+    weights.tFlexMaxLimit = weights.tFlexMaxLimitNoBleed - deltaBleed;
+
+    return weights as LimitWeight;
+  }
+
+  /**
+   * Determine which of the factors is limiting the takeoff weight most for a given temperature.
+   * @param temp The temperature to check.
+   * @param result The partially calculated result.
+   * @returns The most limiting factor.
+   */
+  private getLimitingFactor(
+    temp: 'oatLimit' | 'tRefLimit' | 'tMaxLimit' | 'tFlexMaxLimit',
+    result: Partial<TakeoffPerformanceResult>,
+  ): LimitingFactor {
+    if (!result.limits) {
+      throw new Error('Invalid result object!');
+    }
+
+    let limitingWeight = Infinity;
+    let limitingFactor = LimitingFactor.Runway;
+
+    for (const factor of Object.values(LimitingFactor) as LimitingFactor[]) {
+      const weights = result.limits[factor];
+      if (weights !== undefined && weights[temp] < limitingWeight) {
+        limitingWeight = weights[temp];
+        limitingFactor = factor;
+      }
+    }
+
+    return limitingFactor;
+  }
+
+  /** @inheritdoc */
+  public calculateTakeoffPerformanceOptConf(
+    tow: number,
+    forwardCg: boolean,
+    tora: number,
+    slope: number,
+    lineupAngle: LineupAngle,
+    wind: number,
+    elevation: number,
+    qnh: number,
+    oat: number,
+    antiIce: TakeoffAntiIceSetting,
+    packs: boolean,
+    forceToga: boolean,
+    runwayCondition: RunwayCondition,
+    cg?: number,
+    out?: Partial<TakeoffPerformanceResult>,
+  ): TakeoffPerformanceResult {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const results = [1, 2, 3].map((conf, cacheIndex) =>
+      this.calculateTakeoffPerformance(
+        tow,
+        forwardCg,
+        conf,
+        tora,
+        slope,
+        lineupAngle,
+        wind,
+        elevation,
+        qnh,
+        oat,
+        antiIce,
+        packs,
+        forceToga,
+        runwayCondition,
+        cg,
+        // A320251NTakeoffPerformanceCalculator.optResultCache[cacheIndex],
+      ),
+    );
+
+    const filteredResults = results.filter((r) => r.error === TakeoffPerfomanceError.None);
+
+    // if all the results failed, return the highest conf
+    if (filteredResults.length === 0) {
+      return A320251NTakeoffPerformanceCalculator.deepCopy(results[results.length - 1], out);
+    }
+
+    // pick the result with the highest flex temp, and the lowest speeds (if two have the same flex)
+    filteredResults.sort((a, b) => (a.flex === b.flex ? (a.v1 ?? 0) - (b.v1 ?? 0) : (b.flex ?? 0) - (a.flex ?? 0)));
+    return A320251NTakeoffPerformanceCalculator.deepCopy(filteredResults[0], out ?? {});
+  }
+
+  private static deepCopy(
+    result: TakeoffPerformanceResult,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    out?: Partial<TakeoffPerformanceResult>,
+  ): TakeoffPerformanceResult {
+    // FIXME implement properly
+    return JSON.parse(JSON.stringify(result));
```

Comment:
May want to elaborate on what properly means in this context :P

Reply:
A proper deep copy that doesn't just serialise and de-serialise. 💀